### DISCUSSION TITLE
Blog post on regex constraint decoding

### DISCUSSION THREAD

**Discussion by vivien000** [guidance-ai/guidance#653] [Show and tell]
Hi! You might be interested in this new [blog post](https://vivien000.github.io/blog/journal/llm-decoding-with-regex-constraints.html) in which I describe a new constrained decoding method which ensures both compliance with a regex constraint and "proper" tokenization (all generated token sequences correspond to tokenized strings). The aim of guaranteeing proper tokenization is to increase the decoding speed and reduce the divergence from the probability distribution of the unconstrained LLM. Here is a [Twitter thread](https://x.com/vivien000000/status/1760933100798767314?s=20) with a TL;DR.

**Comment by hudson-ai** [comment]
Very cool, thank you for sharing!
